experimental/segmenter/src/line_breaker.rs

```diff
@@ -244,10 +244,21 @@ fn get_break_state(left: u8, right: u8) -> i8 {
 
 #[inline]
 fn use_complex_breaking_utf32(codepoint: u32) -> bool {
-    // Thai
-    (0xe01..=0xe7f).contains(&codepoint) ||
-    // Burmese
-    (0x1000..=0x109f).contains(&codepoint)
+    let line_break_property = get_linebreak_property_utf32_with_rule(
+        codepoint,
+        LineBreakRule::Strict,
+        WordBreakRule::Normal,
+    );
+    if line_break_property == SA {
+        // Thai
+        (0xe01..=0xe7f).contains(&codepoint) ||
+        // Burmese
+        (0x1000..=0x109f).contains(&codepoint) ||
+        (0xa9e0..=0xa9ff).contains(&codepoint) ||
+        (0xaa60..=0xaa7f).contains(&codepoint)
```

Comment:
Nit: If Shane's nit [in previous comment ](https://github.com/unicode-org/icu4x/pull/1045#discussion_r712432252) is difficult to do, how about we leave a comment here saying something like "The code point ranges define here should be in sync with `get_language()` in lstm.rs."? Similarly, leave a comment in get_language() pointing to use_complex_breaking_utf32(), too

Reply:
I will move this to common function as `language.rs`.